modules/rostests/apitests/shell32/ShellExecuteW.cpp

```diff
@@ -2,178 +2,114 @@
-    else
-        hInstance = ShellExecuteW(NULL, NULL, L"C:\\Windows", NULL,
-                                  NULL, SW_SHOWNORMAL);
-
-    retval = (UINT_PTR) hInstance;
-
-    printf("Return Value for Open %s is '%d'.\n", IsReactOS ? "C:\\ReactOS" : "C:\\Windows", retval);
-
-    ok(retval > 31, "ShellExecuteEx lpFile %s failed. Error: %lu\n",
-                     wine_dbgstr_w(Name), GetLastError());
 
+    // TEST #3: Open Windows folder
+    hInstance = ShellExecuteW(NULL, NULL, WinDir, NULL, NULL, SW_SHOWNORMAL);
+    ret = (INT)(UINT_PTR)hInstance;
+    ok(ret > 31, "TEST #3: ret:%d, LastError: %ld\n", ret, GetLastError());
+    trace("TEST #3 ret: '%d'.\n", ret);
     if (hInstance) 
     {
-        LPCWSTR lpWinClass = L"CabinetWClass";
-        LPCWSTR lpWinTitleWindows = L"C:\\Windows";
-        LPCWSTR lpWinTitleReactOS = L"C:\\ReactOS";
-
-        Sleep(1000);
-        if (IsReactOS)
-            hWnd = FindWindowW(lpWinClass, lpWinTitleReactOS);
-        else
-            hWnd = FindWindowW(lpWinClass, lpWinTitleWindows);
-        PostMessage(hWnd, WM_SYSCOMMAND, SC_CLOSE, 0);  // Terminate Window
+        Sleep(WAIT_SLEEP);
+        // Terminate Window
+        hWnd = FindWindowW(L"CabinetWClass", WinDir);
+        PostMessage(hWnd, WM_SYSCOMMAND, SC_CLOSE, 0);
     }
-// End of test #3 - Open C:\Windows
-
-
-//  ShellExecuteW(handle, "open", <fully_qualified_path_to_folder>, NULL, NULL, SW_SHOWNORMAL);
-
-    if(IsReactOS)
-        hInstance = ShellExecuteW(NULL, L"open", L"C:\\ReactOS\\system32", NULL,
-                                  NULL, SW_SHOWNORMAL);
-    else
-        hInstance = ShellExecuteW(NULL, L"open", L"C:\\Windows\\system32", NULL,
-                                  NULL, SW_SHOWNORMAL);
-
-    retval = (UINT_PTR) hInstance;
-
-    printf("Return Value for C:\\...\\system32 is '%d'.\n", retval);
-
-    ok(retval > 31, "ShellExecuteEx lpFile %s failed. Error: %lu\n",
-                     wine_dbgstr_w(Name), GetLastError());
 
+    // TEST #4: Open system32 folder
+    hInstance = ShellExecuteW(NULL, L"open", SysDir, NULL, NULL, SW_SHOWNORMAL);
+    ret = (INT)(UINT_PTR)hInstance;
+    ok(ret > 31, "TEST #4: ret:%d, LastError: %ld\n", ret, GetLastError());
+    trace("TEST #4 ret: %d.\n", ret);
     if (hInstance)
     {
-        LPCWSTR lpWinClass = L"CabinetWClass";
-        LPCWSTR lpWinTitle = L"C:\\Windows\\system32";
-        Sleep(1000);
-        hWnd = FindWindowW(lpWinClass, lpWinTitle);
-        PostMessage(hWnd, WM_SYSCOMMAND, SC_CLOSE, 0);  // Terminate Window
+        Sleep(WAIT_SLEEP);
+        // Terminate Window
+        hWnd = FindWindowW(L"CabinetWClass", SysDir);
+        PostMessage(hWnd, WM_SYSCOMMAND, SC_CLOSE, 0);
     }
-// End of test #4 - Open C:\Windows\system32
-
-
-//  ShellExecuteW(handle, "explore", <fully_qualified_path_to_folder>, NULL, NULL, SW_SHOWNORMAL);
 
+    // TEST #5: Open "C:\\"
     hInstance = ShellExecuteW(NULL, L"explore", L"C:\\", NULL, NULL, SW_SHOWNORMAL);
```

Comment:
Technically `C:\\` is a hardcoded value as well, since nothing proves that ReactOS is always installed in this drive. You should instead use e.g. the drive letter from the string (`SysDir`) obtained by the GetSystemDirectory call...

+ Same comments for the other hardcoded `C:\\` below.

Reply:
OK.